Mathlib/Analysis/Normed/Group/Basic.lean

```diff
@@ -1376,6 +1350,58 @@ alias ⟨_, HasCompactSupport.norm⟩ := hasCompactSupport_norm_iff
 
 end NormedAddGroup
 
+/-! ### `positivity` extensions -/
+
+namespace Mathlib.Meta.Positivity
+
+open Lean Meta Qq Function
+
+/-- Extension for the `positivity` tactic: multiplicative norms are always nonnegative, and positive
+on non-one inputs. -/
+@[positivity ‖_‖]
+def evalMulNorm : PositivityExt where eval {u α} _ _ e := do
+  match u, α, e with
+  | 0, ~q(ℝ), ~q(@Norm.norm $E $_n $a) =>
+    let _seminormedGroup_E ← synthInstanceQ q(SeminormedGroup $E)
+    assertInstancesCommute
+    -- Check whether we are in a normed group and whether the context contains a `a ≠ 1` assumption
+    let o : Option (Q(NormedGroup $E) × Q($a ≠ 1)) := ← do
+      let .some normedGroup_E ← trySynthInstanceQ q(NormedGroup $E) | return none
+      let some pa ← Qq.findLocalDeclWithType? q($a ≠ 1) | return none
```

Comment:
I hope looking at the local context like this doesn't affect any caching that may or may not exist in positivity. (I don't see any.)

This is something that the simp discharger framework has to worry about, when going under binders.

Functions like `evalTriangleRemovalBound` have pre-existing uses of this, so it's likely OK, but I just want to hear confirmation that this is fine.

Reply:
I can confirm that there is no caching in `positivity` as far as I am aware